LayoutTests/inspector/unit-tests/string-utilities.html

```diff
@@ -161,6 +234,41 @@
         }
     });
 
+    suite.addTestCase({
+        name: "String.prototype.StrictModeReturnTypes",
+        description: "Test that String.prototype methods using strict mode return primitive strings, not String objects.",
+        test() {
+            // Without strict mode, `this` inside the method would be boxed into a String object,
+            // causing the return value to be a String object instead of a primitive string.
+            // This test verifies that strict mode is properly used, ensuring primitive string returns.
+
+            InspectorTest.expectEqual(typeof "abc".collapseWhitespace(), "string", "collapseWhitespace should return a primitive string.");
+            InspectorTest.expectEqual(typeof "abc".removeWhitespace(), "string", "removeWhitespace should return a primitive string.");
+            InspectorTest.expectEqual(typeof "abc".escapeCharacters(""), "string", "escapeCharacters should return a primitive string.");
+            InspectorTest.expectEqual(typeof "abc".escapeForRegExp(), "string", "escapeForRegExp should return a primitive string.");
+            InspectorTest.expectEqual(typeof "abc".capitalize(), "string", "capitalize should return a primitive string.");
+            InspectorTest.expectEqual(typeof "abc".insertWordBreakCharacters(), "string", "insertWordBreakCharacters should return a primitive string.");
+            InspectorTest.expectEqual(typeof "abc".removeWordBreakCharacters(), "string", "removeWordBreakCharacters should return a primitive string.");
+            InspectorTest.expectEqual(typeof "abc".toCamelCase(), "string", "toCamelCase should return a primitive string.");
+            InspectorTest.expectEqual(typeof "abc".truncateStart(10), "string", "truncateStart should return a primitive string.");
+            InspectorTest.expectEqual(typeof "abc".truncateMiddle(10), "string", "truncateMiddle should return a primitive string.");
+            InspectorTest.expectEqual(typeof "abc".truncateEnd(10), "string", "truncateEnd should return a primitive string.");
+            InspectorTest.expectEqual(typeof "abc".truncate(10), "string", "truncate should return a primitive string.");
+
+            // Verify equality comparisons work correctly (would fail if String objects were returned).
+            InspectorTest.expectThat("abc".collapseWhitespace() === "abc", "collapseWhitespace result should be strictly equal to primitive string.");
```

Comment:
let's use `expectEqual` so that if this fails we get a better error message

ditto below